demos/applications/nunchuck.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2024 Khalil Estell
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <libhal-input/gamepad/nunchuck.hpp>
+#include <libhal-util/serial.hpp>
+#include <libhal-util/steady_clock.hpp>
+#include <libhal/i2c.hpp>
+
+#include <resource_list.hpp>
+
+void application(resource_list& p_map)
+{
+  using namespace std::chrono_literals;
+  using namespace hal::literals;
+
+  auto& clock = *p_map.clock.value();
+  auto& console = *p_map.console.value();
+
+  hal::print(console, "Demo Application Starting...\n\n");
+  auto& i2c = *p_map.i2c.value();
+  try {
```

Comment:
Now that we know this will work and what we were catching before was an issue with `hal::soft::bit_bang_i2c`, we can remove this try/catch.